### PR TITLE
Return error properly, add Header metrics

### PR DESCRIPTION
**Pull request by lukanus**

### CODE REVIEW

File: pkg/datastore/headerscontroller.go [hunk 1]
```diff
@@ -18,17 +18,20 @@ type HeaderController struct {
...
+	hc := &HeaderController{
 		slotLag:     slotLag,
 		slotTimeLag: slotTimeLag,
-
-		latestSlot: &latestSlot,
-		headers:    make(map[uint64]*IndexedHeaders),
+		latestSlot:  &latestSlot,
+		headers:     make(map[uint64]*IndexedHeaders),
 	}
+	hc.initMetrics()
```
Comment (aratz-lasa) [inline pkg/datastore/headerscontroller.go:33]: the initMetrics was not called before. Why do we call it no?

**Review by aratz-lasa — CHANGES_REQUESTED** [review]
Could we change `log` for `logger` in: https://github.com/blocknative/dreamboat/blob/73c78004fc292f84735699c56fedc49f076571e4/pkg/relay/relay.go#L160

File: pkg/datastore/headerscontroller.go (hunk 1 above)
Comment (lukanus) [reply pkg/datastore/headerscontroller.go:33]: The other ones are static, it's initiated as in the other ones

**Review by aratz-lasa — APPROVED** [review]
LGTM

**Pull request merged**